src/test/java/io/pravega/connectors/hadoop/PravegaInputFormatITCase.java

```diff
@@ -0,0 +1,77 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.pravega.connectors.hadoop;
+
+import io.pravega.client.stream.EventStreamWriter;
+import io.pravega.connectors.hadoop.utils.IntegerSerializer;
+import io.pravega.connectors.hadoop.utils.SetupUtils;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.mapreduce.InputFormat;
+import org.apache.hadoop.mapreduce.InputSplit;
+import org.apache.hadoop.mapreduce.Job;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+
+public class PravegaInputFormatITCase {
+
+    private static final String TEST_SCOPE = "PravegaInputFormatITCase";
+    private static final String TEST_STREAM = "stream";
+    private static final int NUM_SEGMENTS = 3;
+    private static final int NUM_EVENTS = 20;
+
+    /**
+     * Setup utility
+     */
+    private static final SetupUtils SETUP_UTILS = new SetupUtils();
+
+    @Before
+    public void setupPravega() throws Exception {
+        SETUP_UTILS.startAllServices(TEST_SCOPE);
+        SETUP_UTILS.createTestStream(TEST_STREAM, NUM_SEGMENTS);
+        EventStreamWriter<Integer> writer = SETUP_UTILS.getIntegerWriter(TEST_STREAM);
+        for (int i = 0; i < NUM_EVENTS; i++) {
+            CompletableFuture future = writer.writeEvent(i);
+            future.get();
+        }
+    }
+
+    @After
+    public void tearDownPravega() throws Exception {
+        SETUP_UTILS.stopAllServices();
+    }
+
+    @Test
+    public void testGetSplits() throws IOException, InterruptedException {
```

Comment:
What about a test case that exercises reading splits in the presence of scaling? The test case would get splits at different times corresponding to different numbers of segments. The easiest way is probably to manually scale the stream so that we control the splitting and merging.

Reply:
accepted